modules/hal_exception/src/hal_exception.hpp

```diff
@@ -34,31 +34,30 @@
 
 namespace hal::exception {
 #define ASSERT_WARN(condition)                                               \
-  assert_warn(                                                               \
+  hal::exception::assert_warn(                                               \
```

Comment:
I am a bit confused by the use of the namespace while you are inside of the namespace.
Do you do this because its a macro?

Reply:
Yes, if you use this macro outside of the namespace, normally without adding this namespace it will result in an compiler error. Because the assert_warn will be out of scope. Thus you need to specify the namespace manually causing even more confusion.. Now you can use without any special context use the assert macro's as is without including any namespaces.